lib/matplotlib/tests/test_legend.py

```diff
@@ -1090,3 +1090,37 @@ def test_ncol_ncols(fig_test, fig_ref):
     ncols = 3
     fig_test.legend(strings, ncol=ncols)
     fig_ref.legend(strings, ncols=ncols)
+
+
+def test_loc_short_tuple_exception():
+    # check that exception is raised if the loc arg
+    # of legend is not a 2-tuple of numbers
+    with pytest.raises(ValueError):
+        fig, ax = plt.subplots()
+        ax.legend(loc=(1.1, ))
```

Comment:
Only the `legend` raises, so only it needs to be in the context manager. Also, please add a `match` so that it's certain which exception is being caught.

And I'm pretty sure all three of these invalid tests can be combined into one. Ditto for the valid ones.